cmd/frontend/graphqlbackend/site.go

```diff
@@ -508,24 +508,43 @@ func (r *upgradeReadinessResolver) SchemaDrift(ctx context.Context) ([]*schemaDr
 	return resolvers, nil
 }
 
-// isRequiredOutOfBandMigration returns true if a OOB migration is deprecated not
-// after the given version and not yet completed.
-func isRequiredOutOfBandMigration(version oobmigration.Version, m oobmigration.Migration) bool {
-	if m.Deprecated == nil {
+// isRequiredOutOfBandMigration returns true if an OOB migration will be deprecated in the latest version and has not progressed to completion.
+func isRequiredOutOfBandMigration(currentVersion, latestVersion oobmigration.Version, m oobmigration.Migration) bool {
+	// If current version is dev, no deprecated migrations are required.
+	if currentVersion.Dev && m.Deprecated != nil {
 		return false
 	}
```

Comment:
Based on the code comment, it sounds like we shouldn't check for `m.Deprecated` if `Dev` is true (the "kill switch").
```suggestion
	if currentVersion.Dev {
		return false
	}
```